pkg/monitors/vsphere/service/paginator.go

```diff
@@ -0,0 +1,57 @@
+package service
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/signalfx/signalfx-agent/pkg/monitors/vsphere/model"
+)
+
+// queryPerfPaginator allows callers to split up requests for performance data
+// for all inventory objects into batches. This is because for large enough
+// vSphere deployments, a query for performance data (queryPerf) will fail if
+// performance data is requested for all of the inventory objects in one call.
+type queryPerfPaginator struct {
+	gateway IGateway
+	// The max number of inventory objects requested at a time.
+	// A pagesize of 0 will result in no pagination (one call for all inv objects)
+	pageSize int
+	log      *log.Entry
+}
+
+func (p *queryPerfPaginator) paginate(
+	objs []*model.InventoryObject,
+	maxSample int32,
+) (*types.QueryPerfResponse, error) {
+	numObjs := len(objs)
+	pages := p.numPages(numObjs)
+	start := time.Now()
+	var metrics []types.BasePerfEntityMetricBase
+	for i := 0; i < pages; i++ {
+		startIdx := i * p.pageSize
```

Comment:
I might be missing something, but If pageSize is 0, won't we end up iterating n times (where n is the number of objects in inventory) and end up calling `queryPerf` against the same slice n times?

Reply:
Yeah I think you're right. That was dumb. Will fix.

Reply:
Good catch. Should be fixed.